### PR TITLE
Organize interface of proof

### PR DESCRIPTION
**Pull request by junha1**

### CODE REVIEW

File: src/proof.rs [hunk 1]
```diff
@@ -25,15 +25,15 @@ use primitives::H256;
 #[derive(Clone, Eq, PartialEq, Debug)]
 pub struct CryptoProofUnit {
     pub root: H256,
-    pub key: H256,
+    pub key: Bytes,
```
Comment (majecty) [inline src/proof.rs:28]: Why did you change H256 to Bytes?

Comment (junha1) [reply src/proof.rs:28]: Because all other interfaces of trie take before-hash keys (`&[u8]`)

Comment (majecty) [reply src/proof.rs:28]: Then you need to specify the reason in the commit title or commit message.
I can't find out what did you do in the commit message: "Organize interface of proof"

Also, please split this commit into two commits.
One is changing the key type from H256 to Bytes.
The other one is implementing `CryptoStructure` for the TrieDBMut and make Trie extend CryptoStructure.

They are totally different thing and included in this commit.

**Review by majecty — CHANGES_REQUESTED** [review]

**Comment by majecty** [comment]
Always give the reason why you changed the code.

File: src/proof.rs (hunk 1 above)
Comment (junha1) [reply src/proof.rs:28]: ok

**Comment by majecty** [comment]
@sgkim126 I requested your review since you reviewed the previous Merkle Proof PR and you know well about the rust-merkle-trie library.

File: src/proof.rs (hunk 1 above)
Comment (junha1) [reply src/proof.rs:28]: Can I make 2 commits in **a single PR**? (each commit will have a detailed explanation) @majecty

Comment (majecty) [reply src/proof.rs:28]: Sure.

The main commit will be "implementing the `CryptoStructure` for the TrieDBMut".
The commit that changes key type is preparation work for it.

File: src/proof.rs [hunk 2]
```diff
@@ -127,10 +129,10 @@ mod tests {
     use cdb::MemoryDB;
     use rand::{rngs::StdRng, Rng};
 
-    fn simple_test<'db>(t: &TrieDB<'db>, key: &H256, value: Option<&[u8]>, key_proof: &H256, result: bool) {
+    fn simple_test<'db>(t: &TrieDB<'db>, key: &[u8], value: Option<&[u8]>, key_proof: &[u8], result: bool) {
```
Comment (sgkim126) [inline src/proof.rs]: If it always converts the key to Bytes internally, why not changing the type of the key to Bytes?

Comment (junha1) [reply src/proof.rs]: I'm not sure I understood your point correctly, but the underlying source of such key is a `String` made out of `format!`, not a `Vec`. @sgkim126

**Review by majecty — APPROVED** [review]

File: src/proof.rs (hunk 2 above)
Comment (sgkim126) [reply src/proof.rs]: Even the source of the key is a string, you must and always convert it to a vector internally. In this case, IMO, you can make the typecasting caller's role.

Comment (junha1) [reply src/proof.rs]: I understood what you mean. I updated. Please check. @sgkim126

**Review by sgkim126 — APPROVED** [review]
